lib/topological_inventory/openshift/connection.rb

```diff
@@ -8,16 +8,16 @@ def initialize
       @connection_failure_time = {}
     end
 
-    def connect(endpoint_type, host:, port: 8443, token:, verify_ssl: OpenSSL::SSL::VERIFY_NONE, endpoint_retry_minutes: 60)
+    def connect(endpoint_type, host:, port: 8443, token:, verify_ssl: OpenSSL::SSL::VERIFY_NONE, endpoint_retry_minutes: 60, open_timeout: 10)
```

Comment:
Overall this looks good, I would just ask that we make the default timeout value (10) a constant at the top of this class.  Will make it very easy to identify when looking at the code.